### PR TITLE
SAK-45801 - Calendar: When creating an event in the past or future, u…

### PR DESCRIPTION
**Pull request by mpellicer**
…ser is taken to current week.

### CODE REVIEW

File: calendar/calendar-tool/tool/src/webapp/vm/calendar/chef_calendar_view.vm [hunk 1]
```diff
@@ -22,3 +22,8 @@
 <script src="/library/webjars/fullcalendar/5.8.0/main.min.js"></script>
 <script src="/library/webjars/fullcalendar/5.8.0/locales-all.min.js"></script>
 <script src="/sakai-calendar-tool/js/sakai-calendar.js"></script>
+
+<script>
+  const currentDate = '$currentDate';
+  calendar.gotoDate(currentDate);
```
Comment (adrianfish) [inline calendar/calendar-tool/tool/src/webapp/vm/calendar/chef_calendar_view.vm:28]: You may as well make this one line, unless you want to check that currentDate exists. What happens if you pass undefined or null to gotoDate?

**Review by adrianfish — CHANGES_REQUESTED** [review]

File: calendar/calendar-tool/tool/src/java/org/sakaiproject/calendar/tool/CalendarAction.java [hunk 2]
```diff
@@ -7051,6 +7012,20 @@ protected void buildViewCalendarContext(VelocityPortlet portlet, Context context
 			}
 		}
 
+		Time m_time = TimeService.newTime();
```
[the remark this answers is not in the document]
Comment (mpellicer) [reply calendar/calendar-tool/tool/src/java/org/sakaiproject/calendar/tool/CalendarAction.java:7015]: This is logic that was removed and restored in the backend, the intention of the full calendar work is not update this service in the backend :(

File: calendar/calendar-tool/tool/src/java/org/sakaiproject/calendar/tool/CalendarAction.java [hunk 3]
```diff
@@ -7051,6 +7012,20 @@ protected void buildViewCalendarContext(VelocityPortlet portlet, Context context
...
+		TimeBreakdown b = m_time.breakdownLocal();
```
[the remark this answers is not in the document]
Comment (mpellicer) [reply calendar/calendar-tool/tool/src/java/org/sakaiproject/calendar/tool/CalendarAction.java:7016]: This is logic that was removed and restored in the backend, the intention of the full calendar work is not update this service in the backend :(

File: calendar/calendar-tool/tool/src/webapp/vm/calendar/chef_calendar_view.vm (hunk 1 above)
Comment (mpellicer) [reply calendar/calendar-tool/tool/src/webapp/vm/calendar/chef_calendar_view.vm:28]: If you pass a non-parsable value it uses the browser current date!

File: calendar/calendar-tool/tool/src/java/org/sakaiproject/calendar/tool/CalendarAction.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (mpellicer) [reply calendar/calendar-tool/tool/src/java/org/sakaiproject/calendar/tool/CalendarAction.java:7015]: Okay, I've replaced the snippet using ZonedDateTime, thanks!

File: calendar/calendar-tool/tool/src/java/org/sakaiproject/calendar/tool/CalendarAction.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (mpellicer) [reply calendar/calendar-tool/tool/src/java/org/sakaiproject/calendar/tool/CalendarAction.java:7016]: Okay, I've replaced the snippet using ZonedDateTime, thanks!

**Review by adrianfish — APPROVED** [review]
GTG. Thanks Miguel.

**Comment by mpellicer** [comment]
Thanks to you!

**Pull request merged**